packages/app-design-forest/src/customPropsTree.ts

```diff
@@ -12,18 +12,33 @@ export type SimpleCustomProp = {
     | "array_static_asset"
     | "internal_link"
     | "component_selector"
-    | "map"
     | "external_link";
 };
 
+export type MapCustomProp = {
+  type: "map";
+  attributes: {
+    name: string;
+    type:
```

Comment:
Aren't we expecting all `SimpleCustomProp` here. If you think you will be able to provide support for all `SimpleCustomProp` then just put here `type: SimpleCustomProp`.

I guess you have excluded `array` types only, but, I think we should support those as well as `EnumCustomProp` here. Hence, the `type` here should be something like `type: SimpleCustomProp | EnumCustomProp`

Reply:
Done. Yes this makes a lot more sense than my previous approach